pkg/queryfrontend/cache/memcached.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package cache
+
+import (
+	"context"
+	"time"
+
+	cortexcache "github.com/cortexproject/cortex/pkg/chunk/cache"
+	"github.com/cortexproject/cortex/pkg/querier/queryrange"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"gopkg.in/yaml.v2"
+
+	"github.com/thanos-io/thanos/pkg/cacheutil"
+)
+
+const (
+	memcachedDefaultTTL = 24 * time.Hour
+)
+
+// MemcachedCache is a memcached-based cache.
+type MemcachedCache struct {
+	logger    log.Logger
+	memcached cacheutil.MemcachedClient
+	validity  time.Duration
+
+	// Metrics.
+	requests prometheus.Counter
+	hits     prometheus.Counter
+}
+
+type MemcachedResponseCacheConfig struct {
+	Memcached cacheutil.MemcachedClientConfig `yaml:",inline"`
+	Validity  time.Duration                   `yaml:"validity"`
+}
+
+// newMemcachedCache makes a new MemcachedCache and returns Cortex ResultsCacheConfig.
+func newMemcachedCache(conf []byte, logger log.Logger, reg prometheus.Registerer) (*queryrange.ResultsCacheConfig, error) {
+	var config MemcachedResponseCacheConfig
+	if err := yaml.UnmarshalStrict(conf, &config); err != nil {
+		return nil, err
+	}
+
+	memcachedClientConfig, err := yaml.Marshal(config.Memcached)
```

Comment:
Why are we re-marshaling the inlined config.Memcached field? We could directly pass `conf` to the `cacheutil.NewMemcachedClient` func and save this extra step, no?

Reply:
:+1:

Reply:
You are right! Thanks.